src/damage.cpp

```diff
@@ -74,31 +74,19 @@ bool damage_evasion(CharData *ch, CharData *attacker, ObjData *weapon, int dtype
             s = std::max(susceptibility(ch, skill_to_dtype(GET_OBJ_VAL(weapon, VAL_WEAPON_DAM_TYPE) + TYPE_HIT)),
                          susceptibility(ch, DAM_FIRE));
         if (OBJ_EFF_FLAGGED(weapon, EFF_ICE_WEAPON)) {
```

Comment:
The way this is currently setup, if a weapon is flagged as a poison and acid weapon, for example, it's going to trigger multiple of these.  I don't think we should allow that. 

While it's not really a big deal, I'd prefer these turn into `else if` statements.  mainly so we don't have to check each if statement each time.  It'll make it perform slightly better.

Also, we should consider what would happen if a weapon had multiple damage types.

Reply:
Actually, I had written the code to account for that originally and forgot what I had been thinking, so I've restored it.  Line 124 is the actual bug fix.

The idea is if a weapon has multiple energy type flags, it should check every flag and compare for the most favorable one when considering _evasion._  A water elemental might have a good chance to evade a flaming bludgeon, but if it's a _holy_ flaming bludgeon, it shouldn't have a chance to dodge.

Note this check does _not_ apply to damage.  I don't know how to handle multiple damage types.  I had started to work it out, but I couldn't figure out how to divide up the total damage dynamically and then send each type through the damage function.

Reply:
I'll push this for now, but sounds like there's more work to do.